slack-pi-command/README.md

```diff
@@ -0,0 +1,171 @@
+
+```diff
++const { parse } = require('query-string');
+
+ exports.handler = async (body, context) => {
++  const parsed = parse(context.body.toString('utf8'));
+-  const name = context.request.query.name || body.name || 'World';
+-  return `Hello ${name}!`;
+ };
+```
+
+Now that we have the payload in a usable form, let's extract the data we need, so we compute π to the number of digits requested from our user.
+
+```diff
+   const parsed = parse(context.body.toString('utf8'));
++  const numDigits = parseInt(parsed.text, 10);
+ };
+```
+
+Now let's add some generic validation
+
+```diff
+   const numDigits = parseInt(parsed.text, 10);
++
++  // ensure the input is a valid, natural number
++  if (isNaN(numDigits) || numDigits <= 0) {
++    return {
++      text: `Expected non-zero, natural number, received "${parsed.text}"`,
++      response_type: 'ephemeral',
++    };
++  }
+
++  // enforce a fun limit on the max pi digits
++  if (numDigits > 10) {
++    return {
++      text: "You've already had too much 🥧!",
++      response_type: 'ephemeral',
++    };
++  }
+ };
+```
+
+> Note: We return the specific fields `text` & `response_type` because this is the expected format of an "Error" response in the Slack API.
+
+All that's left to do is return π to the number of digits requested by the user. The `npm` package `pi` makes this very convenient.
+
+
+```diff
+   }
++  return PI(numDigits - 1);
+};
+
+```
+
+> Note: We subtract one from the input value because the first 3 in π is not considered a "digit"
+
+Don't forget to `require` the package at the top
+
+```diff
+ const { parse } = require('query-string');
++const PI = require('pi');
+
+ exports.handler = async (body, context) => {
+```
+
+The final state of `function.js`
+
+```JavaScript
+const { parse } = require('query-string');
+const PI = require('pi');
+
+exports.handler = async (body, context) => {
+  // slack unfortunately sends command payloads as x-url-form-encoded
+  const parsed = parse(context.body.toString('utf8'));
+
+  // ensure the input is a valid, natural number
+  const numDigits = parseInt(parsed.text, 10);
+  if (isNaN(numDigits) || numDigits <= 0) {
+    return {
+      text: `Expected non-zero, natural number, received "${parsed.text}"`,
+      response_type: 'ephemeral',
+    };
+  }
+
+  // enforce a fun limit on the max pi digits
+  if (numDigits > 10) {
+    return {
+      text: "You've already had too much 🥧!",
+      response_type: 'ephemeral',
+    };
+  }
+  return PI(numDigits - 1);
+};
+```
+
+Finally, let's install the dependencies we used and deploy our function (so it's invokable over HTTPS).
+
+```bash
+$ npm install --save query-string pi
+$ bn deploy public_slack_pi_command
+```
+
+Make sure to copy and save the URL printed by `bn deploy`, we will use this URL during the next step
+
+```bash
+# example URL
+https://run.binaris.com/v2/run/0123456789/public_slack_pi_command
```

Comment:
maybe attach this section to the `bn deploy` (and separate it from `npm install`).
it will eliminate the need to the bash comment which is not really printed and might confuse